partner_contact_depth/models/res_partner.py

```diff
@@ -0,0 +1,38 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.depends("parent_id")
+    def _compute_contact_depth(self):
+        for partner in self:
+            depth = 0
+            parent = partner.parent_id
+            while parent:
+                depth += 1
+                parent = parent.parent_id
+            partner.depth = depth
+
+    depth = fields.Integer(compute="_compute_contact_depth", store=True)
+
+    @api.constrains("depth")
+    def check_maximum_contact_depth(self):
+        max_depth = int(
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("partner_contact_depth.contacts_max_depth", default=0)
+        )
+        for partner in self:
+            if 0 < max_depth < partner.depth:
+                raise ValidationError(
+                    _(
+                        "You have reached the maximum depth of contacts. "
+                        "You cannot create childs for this contact."
+                    )
+                )
+
+    def write(self, vals):
+        self.check_maximum_contact_depth()
+        return super().write(vals)
```

Comment:
Is this really needed? I guess that, as the `depth` will be a stored field, updated every time there is a relevant change, we should already be covered by that.
WDYT?

Reply:
The aim is to avoid contacts modification with depth higher that the maximum defined.